vmicore/src/lib/plugins/PluginSystem.cpp

```diff
@@ -254,14 +242,26 @@ namespace VmiCore
         }
     }
 
-    void PluginSystem::passShutdownEventToRegisteredPlugins()
+    void PluginSystem::unloadPlugins()
     {
         vmiInterface->flushV2PCache(LibvmiInterface::flushAllPTs);
         vmiInterface->flushPageCache();
 
-        for (auto& shutdownCallback : registeredShutdownCallbacks)
+        for (auto& plugin : plugins)
```

Comment:
I think I would move both plugin-init and unloading into the PluginSystem. In the current version you have the init-loop over all plugins in VmiHup and the unload-loop in PluginSystem. Feels a bit off.